src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -88,48 +86,62 @@ internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
 
         private void WriteLocations(Location location, IReadOnlyList<Location> additionalLocations)
         {
-            _writer.WriteArrayStart("locations");
+            if (location.SourceTree != null)
+            {
+                _writer.WriteArrayStart("locations");
+                _writer.WriteObjectStart(); // location
+                _writer.WriteKey("analysisTarget");
```

Comment:
ok, need @lgolding to pay attention to this one, too. i think you should be populating 'resultFile' here. 'resultFile' is where you provide the result location, except in cases where that specific file is itself the analysis target. for a roslyn compilation, i'd argue no single source file is the analysis target, it's the assembly itself that is under analysis. raises the issue of whether you should populate 'analysisTarget' with that information. i think you have to, otherwise, how could a post-processing tool aggregate issues by individual projects/assemblies?